src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/CoreCLR/WebRequestPSCmdlet.CoreClr.cs

```diff
@@ -389,6 +389,11 @@ internal virtual void FillRequestStream(HttpRequestMessage request)
                     byte[] bytes = content as byte[];
                     SetRequestContent(request, bytes);
                 }
+                else if (content is MultipartFormDataContent multipartFormDataContent)
+                {
+                    WebSession.ContentHeaders.Clear();
```

Comment:
`WebSession.ContentHeader` is set at line 325 when `ContentType` is specified. Is it OK to ignore that as well?

Reply:
Yes, `ContentType` must be set by the `MultipartFormDataContent` object as it controls the `boundary` in the `Content-Type` header used to distinguish between the various fields supplied. Documentation will probably need to make clear that `-ContentType` will be ignored when a `MultipartFormDataContent` object is supplied. It doesn't make sense to use any other `Content-Type` header when doing `multipart/form-data` anyway.  If this isn't done here, the Foreach on line 421 throws. when `Content` is set with a `MultipartFormDataContent` it overwrites all the content headers, so trying to add content headers again causes an exception about `ContentType` already existing.

Reply:
Thanks for the clarification. I updated the PR description to capture your words.

Then there is potential problem at line 239:
```
if (WebSession.Headers.Count > 0)
{
    WebSession.ContentHeaders.Clear();
    foreach (var entry in WebSession.Headers)
    {
        if (HttpKnownHeaderNames.ContentHeaders.Contains(entry.Key))
        {
            WebSession.ContentHeaders.Add(entry.Key, entry.Value);
        }
...
```
If `-Headers` or `-WebSession` is specified, then the `foreach` on line 421 would throw.